### DISCUSSION TITLE
How to add a row to a database that contains an Array with Objects?

### DISCUSSION THREAD

**Discussion by skorotkiewicz** [prisma/prisma#8583] [Q&A] [answered]
It need to add a new row, using upsert() which contains the following data:

```javascript
      let priv = {
        sender,
        recipient,
        nonce,
        read: [{ [sender]: false, [recipient]: false }],
      };

    // And simply add
    await prisma.privMsg.upsert({
      where: { nonce },
      update: privs,
      create: privs,
    });
```

Here is my scheme, I don't really know what type to use for my `read` data.
```prisma
model PrivMsg {
  nonce     String    @id
  sender    String
  recipient String
  read      Boolean[]
//  read      String[]
}
```

<details><summary>Here prisma log</summary>

```log
Invalid `prisma.privMsg.upsert()` invocation:

{
  where: {
    nonce: 'ShZjPjJ4PMNxqDnUO-W7ZNHJ'
  },
  update: {
    sender: 'ola',
    recipient: 'aptitude',
    nonce: 'ShZjPjJ4PMNxqDnUO-W7ZNHJ',
    read: [
      {
        ola: false,
        aptitude: false
      }
    ]
    ~~~~~~~~~~~~~~~~~~~
  },
  create: {
    sender: 'ola',
    recipient: 'aptitude',
    nonce: 'ShZjPjJ4PMNxqDnUO-W7ZNHJ',
    read: [
      {
        ola: false,
        aptitude: false
      }
    ]
    ~~~~~~~~~~~~~~~~~~~
  }
}

Argument read: Got invalid value 
[
  {
    ola: false,
    aptitude: false
  }
]
on prisma.upsertOnePrivMsg. Provided List<Json>, expected PrivMsgUpdatereadInput or Boolean:
type PrivMsgUpdatereadInput {
  set?: List<Boolean>
  push?: Boolean | List<Boolean>
}
type PrivMsgUpdatereadInput {
  set?: List<Boolean>
  push?: Boolean | List<Boolean>
}
Argument read: Got invalid value 
[
  {
    ola: false,
    aptitude: false
  }
]
on prisma.upsertOnePrivMsg. Provided List<Json>, expected PrivMsgCreatereadInput or Boolean:
type PrivMsgCreatereadInput {
  set: List<Boolean>
}
type PrivMsgCreatereadInput {
  set: List<Boolean>
}

```
</details>

**Comment by ryands17** [comment, marked as answer]
@skorotkiewicz 👋 

As mentioned in the error, you either need to use `set` or `push` in `read`. `read` is just an array of Booleans so just passing `true` or `false` like this should work:

```ts
let priv = {
        sender,
        recipient,
        nonce,
        read: { set: [false, false] },
};

    // And simply add
await prisma.privMsg.upsert({
      where: { nonce },
      update: privs,
      create: privs,
});
```

**Reply by skorotkiewicz** [reply]
But this only save Booleans value, but a need to save object keys too.

This is user names:
`{ [sender]: false, [recipient]: false }`
It looks like:
`{userName1: false, userName2: false}`

**Reply by ryands17** [reply]
What you have added is `read      Boolean[]`. This only saves booleans. For object keys, you either create a new model with a relation or use [Json](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
